src/database/migrations/20241217112401-update-reports-queries-data.js

```diff
@@ -136,38 +136,40 @@ module.exports = {
                     OR :session_type = 'Public' AND s.type = 'PUBLIC'
                     OR :session_type = 'Private' AND s.type = 'PRIVATE'
                 )
-                -- Search logic for recommended_for or specific columns
-                AND (
-                    :search_column IS NULL OR (
-                        LENGTH(:search_value) >= 1 AND 
-                        (
-                            -- Search within specific columns
-                            (
-                                :search_column = 'sessions_title' AND s.title ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'sessions_created_by' AND ue.name ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'mentor_name' AND s.mentor_name ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'date_of_session' AND TO_CHAR(TO_TIMESTAMP(s.start_date)::DATE, 'YYYY-MM-DD') ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'session_type' AND s.type ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'categories' AND ARRAY_TO_STRING(s.categories, ', ') ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'recommended_for' AND ARRAY_TO_STRING(s.recommended_for, ', ') ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'duration_of_sessions_attended' AND 
-                                ROUND(EXTRACT(EPOCH FROM (TO_TIMESTAMP(s.end_date) - TO_TIMESTAMP(s.start_date))) / 60)::TEXT ILIKE '%' || :search_value || '%'
-                            ) OR (
-                                :search_column = 'session_attended' AND (
-                                    (:search_value = 'Yes' AND sa.joined_at IS NOT NULL)
-                                    OR (:search_value = 'No' AND sa.joined_at IS NULL)
-                                )
-                            )
+                -- Filter by a specific column and value
+                AND (:filter_column IS NULL OR 
+                (
+                    (:filter_column = 'sessions_title' AND s.title = :filter_value)
+                    OR (:filter_column = 'sessions_created_by' AND ue.name = :filter_value)
```

Comment:
@sumanvpacewisdom its should be and condition right

Reply:
@sumanvpacewisdom can we construct the filter and sort and search from the input ?